cucumber/policy/features/support/policy_helpers.rb

```diff
@@ -111,43 +69,30 @@ def json_result
     end
   end
 
-  def header(token = nil)
-    if token.nil?
-      token = admin_token
-    end
-    { Authorization:  %Q(Token token="#{token}") }
-  end
-
-  def uri(root, kind, id = nil)
-    uri = "#{appliance_url}/#{root}/#{account}/#{CGI.escape(kind)}"
-    return uri if id.nil?
-
-    "#{uri}/#{CGI.escape(id)}"
-  end
-
-  def admin_password
-    'SEcret12!!!!'
-  end
-
-  def appliance_url
-    ENV['CONJUR_APPLIANCE_URL'] || 'http://conjur'
+  def make_full_id(*tokens)
+    super(tokens.join(":"))
   end
 
-  def account
-    ENV['CONJUR_ACCOUNT'] || 'cucumber'
+  def rest_client(root, kind, id)
+    ClientHelpers::Client.new(root, kind, id)
   end
 
   def login_as_role(login, api_key = nil)
-    api_key = admin_api_key if login == "admin"
+    client = rest_client("authn","login","any")
+    api_key = client.admin_api_key if login == "admin"
     unless api_key
       role = if login.index('/')
         login.split('/', 2).join(":")
       else
         [ "user", login ].join(":")
       end
-      api_key = create_api_key(role)
+      api_key = client.create_api_key(role)
+    end
+    if login == "admin"
+      @token = client.admin_token
+    else
+      @token = client.get_token(login, api_key)
```

Comment:
Let's discuss this method too... It's existing, but maybe we can clean it up more now.... I haven't thought it through yet.